randobot/handler.py

```diff
@@ -161,13 +171,521 @@ async def race_data(self, data):
+            # Handle selecting setting from different pool.
+            if len(args) == 2 and args[0] in major_pool.keys():
+                await self.send_message(
+                    'Invalid pool. Use !settings for available options.'
+                )
+                return
+            elif len(args) == 2 and args[0] in minor_pool.keys():
+                if args[1] in minor_pool.get(args[0]).keys():
+                    await self.send_message(
+                        f'{args[0].capitalize()} will be set to: {args[1].capitalize()}'
+                    )
+                    # Move setting keyword from available pool to picks pool. Add literal setting to data pool.
+                    picks.update({args[0]: args[1]})
+                    data.update({
+                        setting[0]: setting[1] for setting in minor_pool.get(args[0]).get(args[1]).items()
+                    })
+                    minor_pool.pop(args[0])
+                    # Advance draft state. Update status after final pick.
+                    draft['pick_count'] += 1
+                else:
+                    await self.send_message(
+                        f'Invalid option for {args[0].capitalize()}. Available options are: {", ".join(value for value in minor_pool.get(args[0]).keys())}'
+                    )
+                    return
+                if draft.get('pick_count') == 4:
+                    draft.update({
+                        'status': 'complete',
+                        'current_selector': None
+                    })
+                    await self.send_message(
+                        'All picks have been recorded.'
+                    )
+                    await self.send_message(
+                        'Use !seed 15 minutes prior to race start to roll the seed.'
+                    )
+                    return
+                # Change player turn post setting selection.
+                if reply_to == racer[0].get('name'):
+                    draft.update({'current_selector': racer[1].get('name')})
+                elif reply_to == racer[1].get('name'):
+                    draft.update({'current_selector': racer[0].get('name')})
+                await self.send_message(
+                    f"{draft.get('current_selector')}, modify a minor setting with !pick <setting> <value>."
+                )
+                await self.send_message(
+                    'Use !settings to for available options.'
+                )
+                return
+            # Handle invalid format and unknown arguments.
+            await self.send_message(
+                'Invalid option. Use !settings for available options.'
+            )
+
+    async def ex_settings(self, args, message):
+        """
+        Handle !settings commands.
+
+        List available settings and values.
+        """
+        draft = self.state.get('draft_data')
```

Comment:
Actually one more thing - we need an `if not draft.get('enabled')` check here - otherwise the line below will error if `!settings` is used outside of draft.